### PR TITLE
Convert add dependent modal to Vue 3 for AB#15820.

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#15820](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15820)

## Description

- Converted new dependent component to VUE 3
- Added link to Add Dependent Dialog to Dependents 
- Updated functional tests as data-testid have been updated for some tags

**NOTE: DependentViewSelectorComponent.vue will be updated again via [AB#15821](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15821) which is currently being worked on.  Button has been added to test functionality.**

## Testing

- [ ] Unit Tests Updated
- [x] Functional Tests Updated
- [ ] Not Required

## UI Changes

**Link from Dependents to Dialog:**

Screenshot 2023-07-27 at 11 02 19 PM

**Register button disabled when form validation fails:**

RegisterButtonDisabled

**Register button enabled after entering fields:**

Screenshot 2023-07-28 at 9 27 36 AM

**429 Error:**

429Error

**Data Mismatch Error:**

DataMismatch

## Notes

There seems to be a bug with the PageTitleComponent.  Both AddQuickLInkComponent  and AddDependentComponent have issues when the v-dialog is large when used in mobile mode like iPhone 4.  You are unable to scroll down on the page to see the complete v-dialog.  Created bug [AB#15871](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15871)

**iPhone 6/7/8 Mode Issue with PageTitleComponent:**

Screenshot 2023-07-27 at 8 48 25 PM

Screenshot 2023-07-27 at 10 53 28 PM

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by BronzBierd — CHANGES_REQUESTED** [review]
I agree with the data-testid changes that would be better in line with the kebab-case convention however, Mike has set the convention if we're changing the casing to change it in both the new and the old component to allow for backwards compatibility with the Older ClientApp code for interim coverage.

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+            ),
+        },
+    },
+    accepted: { isChecked: sameAs(true) },
+}));
+
+const v$ = useVuelidate(validations, { dependent, accepted });
+
+function addDependent(): void {
+    isLoading.value = true;
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:144]: This is a good component to use the loadingStore.

which works well with a single loading state. I haven't used it on the timeline components as we would be registering a dozen loaders which would trigger the loading event for a multitude of components.

this could be argued as a limitation of using enum type for loaders. however in Single loader components like page/form/non-repeating components it is a a good place to use the loadingStore.

https://github.com/bcgov/healthgateway/blob/features/15545/Apps/WebClient/src/NewClientApp/src/components/private/profile/ProfileView.vue has a good example of using the loadingStore.

In short you register to the loader through a computed property and then use the helper function to manage promises that should activate the loader, and bob's your uncle.

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+        .finally(() => (isLoading.value = false));
+}
+
+function clear(): void {
+    dependent.value = {
+        firstName: "",
+        lastName: "",
+        dateOfBirth: "",
+        PHN: "",
+    };
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:175]: The empty dependent value is used twice, this can be moved to its on variable for reuse

```
const emptyDependent = { 
    //... your model here 
}

dependent.value = {...emptyDependent}
```

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+    <v-dialog
+        id="add-dependent-dialog"
+        v-model="isVisible"
+        data-testid="add-dependent-dialog"
+        scrollable
+        width="700px"
+        persistent
+        no-click-animation
+    >
+        <template #activator="slotProps">
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:213]: NO WORK, but need a decision.

I see there's some inconsistency in the implementations. We'll need to decide on a convention here.

I prefer `#activator="{ props: slotProps }"` for the definition, otherwise we'll need to consider something more "complete" if we want to get the entire value passed through like `#activator="slotObj"` and then consume as you have through `slotObj.props`

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+                                class="text-body-1"
+                            >
+                                If you continue to have issues, please contact
+                                <a href="mailto:HealthGateway@gov.bc.ca"
+                                    >HealthGateway@gov.bc.ca</a
+                                >.
+                            </span>
+                        </template>
+                    </v-alert>
+                    <v-container data-testid="newDependentModalText">
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:297]: this testid should be kebab-case for consistenc

I don't really think the v-container is necessary. If it helped with the layout though feel free to leave it.

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+                        <v-row>
+                            <v-col cols="12" sm="6">
+                                <label for="firstName"
+                                    >First and Middle Names</label
+                                >
+                                <div>
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:303]: div is unnecessary here, v-text-field has display: block, thus it takes up the whole row.

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+                                                v$.dependent.firstName
+                                            )
+                                        "
+                                        @blur="v$.dependent.firstName.$touch()"
+                                    />
+                                </div>
+                            </v-col>
+                            <v-col cols="12" sm="6">
+                                <label for="lastName">Last Name</label>
+                                <div>
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:323]: simliar no need for the divs, please see other instances below

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue [hunk 7]
```diff
@@ -0,0 +1,457 @@
...
+                                    />
+                                </div>
+                            </v-col>
+                        </v-row>
+                        <v-row>
+                            <v-col>
+                                <v-checkbox
+                                    id="termsCheckbox"
+                                    v-model="accepted"
+                                    data-testid="termsCheckbox"
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:386]: another testId that could be changed to kebab-case

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue [hunk 8]
```diff
@@ -0,0 +1,457 @@
...
+                                    color="primary"
+                                    class="text-body-1 checkbox-top"
+                                    hide-details
+                                    :error="
+                                        !ValidationUtil.isValid(v$.accepted)
+                                    "
+                                    @blur="v$.accepted.$touch()"
+                                >
+                                    <template #label>
+                                        <div class="ml-1">
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:397]: NIT: I'll leave this to your discretion:

I doubt the div is really necessary here too, similar to above, p is a block element and should take the entire width already. The margin class is unusual as the checkbox styling should already be managed by vuetify.

**Review by BronzBierd — CHANGES_REQUESTED** [review]
Some minor changes, and a request to use the loadingStore for the consistency.

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue (hunk 1 above)
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:144]: Changed.

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue (hunk 4 above)
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:297]: Removed v-container.

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue (hunk 8 above)
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:397]: No change.  Div makes a difference.

Screenshot 2023-07-28 at 2 14 29 PM

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue (hunk 2 above)
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:175]: Changed.

File: Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue [hunk 9]
```diff
@@ -0,0 +1,445 @@
...
+const dependentStore = useDependentStore();
+const errorStore = useErrorStore();
+const loadingStore = useLoadingStore();
+const userStore = useUserStore();
+
+const isVisible = ref(false);
+const errorMessage = ref("");
+const errorType = ref<ActionType | null>(null);
+const accepted = ref(false);
+const dependent = ref<AddDependentRequest>(emptyDependent);
```
Comment (BronzBierd) [inline Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue:74]: This should be a copy of `emptyDependent`, so that we don't mutate the `emptyDependent` values
```suggestion
const dependent = ref<AddDependentRequest>({ ...emptyDependent });
```

**Review by BronzBierd — CHANGES_REQUESTED** [review]
